nutanix/data_source_nutanix_pbr.go

```diff
@@ -0,0 +1,428 @@
+										Computed: true,
+									},
+									"vpc_reference": {
+										Type:     schema.TypeMap,
+										Computed: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+									"destination": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"address_type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"source": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"address_type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"priority": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"protocol_parameters": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"tcp": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"destination_port_range_list": portRangeSchema(),
+															"source_port_range_list":      portRangeSchema(),
+														},
+													},
+												},
+												"udp": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"destination_port_range_list": portRangeSchema(),
+															"source_port_range_list":      portRangeSchema(),
+														},
+													},
+												},
+												"icmp": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"icmp_type": {
+																Type:     schema.TypeInt,
+																Computed: true,
+															},
+															"icmp_code": {
+																Type:     schema.TypeInt,
+																Computed: true,
+															},
+														},
+													},
+												},
+												"protocol_number": {
+													Type:     schema.TypeInt,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"action": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"action": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"protocol_type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
```

Comment:
If this field can only be set with certain limited values then we can add a validator here to throw error incase user gives wrong value. Check this https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#:~:text=%C2%BB-,ValidateDiagFunc,-Data%20structure%3A

Reply:
validator is there. Check latest commit .